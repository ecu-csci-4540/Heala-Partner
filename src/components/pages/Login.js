import React, { useCallback } from "react";
import PropTypes from "prop-types";
import { useAlert } from "components/hooks";
import { ReactComponent as HealaIconW } from "assets/images/logo-white1.svg";
import { useHistory } from "react-router-dom";
import { Grid, Avatar } from "@mui/material";
import vec from "assets/images/vec.png";
import { makeStyles } from "@mui/styles";

// import { useSelector } from 'react-redux'
import { Login_USER } from "components/graphQL/Mutation";
import { useMutation } from "@apollo/client";
import { setAccessToken } from "../../accessToken";
import { useActions } from "components/hooks/useActions";
import { LoginForm } from "components/pages/forms";
const useStyles = makeStyles((theme) => ({
  // form: theme.mixins.toolbar,
  background: {
    width: "100%",
    minHeight: "100vh !important",
    background:
      "linear-gradient(0deg, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2)),linear-gradient(98.44deg, #3e5ea9 1.92%, #7eedba 122.04%) !important",
  },
  secV: {
    backgroundImage: `url(${vec})`,
    opacity: " 0.05",
    width: "100%",
    height: "100.1%",
    position: "absolute",
    backgroundRepeat: "round",
  },
}));

const Login = () => {
  const classes = useStyles();

  const { displayMessage } = useAlert();
  const history = useHistory();
  const [loginInfo] = useMutation(Login_USER);
  const { loginUser, loginFailue } = useActions();

  const onSubmit = useCallback(async (values, onSubmitProps) => {
    try {
      const { email, password, authType } = values;
      const { data } = await loginInfo({
        variables: {
          data: { email, password, authType },
        },
      });
      if (data?.login?.account?.role === "partner") {
        const { email, _id, access_token, providerId } = data?.login?.account;
        setAccessToken(access_token);
        localStorage.setItem("AppId", _id);
        localStorage.setItem("partnerProviderId", providerId);
        localStorage.setItem("AppEmail", email);
        loginUser({
          data,
          messages: {
            message: "Login successful",
            type: "success",
          },
        });
        displayMessage("success", "Login successful");
        history.push("/dashboard");
      } else {
        displayMessage("error", "Please login using a Partner account");
      }
      onSubmitProps.resetForm({
        values: {
          ...values,
          password: "",
        },
      });
    } catch (error) {
      console.error(error);
      displayMessage("error", "Failed to login");
      loginFailue({
        message: error.message,
        type: "error",
      });
    }
    //eslint-disable-next-line
  }, []);
  return (
    <Grid
      container
      justifyContent="center"
      alignItems="center"
      className={classes.background}
    >
      <div className={classes.secV}></div>
      <Grid container justifyContent="center" margin="auto">
        <Grid
          container
          style={{
            marginTop: "-10%",
            justifyContent: "center",
            alignItems: "center",
          }}
        >
          <Avatar
            sx={{
              background: "transparent",
              color: "white",
              width: 150,
              height: 150,
            }}
          >
            <HealaIconW />
          </Avatar>
        </Grid>
        <Grid
          item
          container
          md={5}
          lg={3}
          xs={11}
          direction="column"
          sx={{
            padding: "4rem 3rem 3rem",
            background: "white",
            borderRadius: "5px",
            margin: "auto",
          }}
        >
          <Grid item>
            <LoginForm onSubmit={onSubmit} />
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

Login.propTypes = {
  history: PropTypes.object,
};

export default Login;
