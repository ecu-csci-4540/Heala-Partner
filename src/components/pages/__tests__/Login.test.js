import Login from "../Login";
import React from "react";
// import { ThemeProvider } from "@mui/material/styles";
// import { ThemeProvider } from "@material-ui/core/styles";
import ReactDOM from "react-dom";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);
const mockUseTheme = jest.fn();

jest.mock("@mui/material/styles", () => () => ({
  ...jest.requireActual("@mui/material/styles"),
  useTheme: () => {
    return mockUseTheme;
  },
}));
// jest.mock("@mui/material/styles", () => {
//   const Styles = jest.requireActual("@mui/material/styles");

//   const classes = jest.requireActual("@mui/styles").useStyles();

//   //   const options = jes?St.requireActual("../../../src/themes/options").default;

//   return {
//     ...Styles,

//     makeStyles: (func) => {
//       return Styles.makeStyles(func.bind(null, classes));
//     },
//   };
// });

it("Should render Example component", () => {
  const wrapper = render(<Login />);
  screen.debug();
  expect(wrapper).toBeTruthy();
});
