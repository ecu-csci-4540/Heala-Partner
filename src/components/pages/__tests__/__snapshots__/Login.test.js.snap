// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Should display alert after login 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-background-17 css-1hbmzt3-MuiGrid-root"
  >
    <div
      class="makeStyles-secV-18"
    />
    <div
      class="MuiGrid-root MuiGrid-container css-1pkg14b-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        style="margin-top: -10%; justify-content: center; align-items: center;"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-5crd6t-MuiAvatar-root"
        >
          <svg>
            logo-white1.svg
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11 MuiGrid-grid-md-5 MuiGrid-grid-lg-3 css-1tum5lx-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <form
            action="#"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-ri7bw8-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item css-biog6w-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-sm-10 MuiGrid-grid-md-12 css-1u8whp-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 makeStyles-header-21 css-ag7rrr-MuiTypography-root"
                  >
                    LOGIN TO PARTNER ACCOUNT
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-sm-10 MuiGrid-grid-md-12 css-17y0cpk-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
                      >
                        Email
                      </p>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                      >
                        <div
                          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedStart css-18k917r-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                          </div>
                          <input
                            aria-label="Enter your email"
                            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                            id="email"
                            name="email"
                            placeholder="Enter your email"
                            type="email"
                            value="shally@gmail.com"
                          />
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-hdw1oc"
                            >
                              <span
                                class="notranslate"
                              >
                                ​
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-sm-10 MuiGrid-grid-md-12 css-17y0cpk-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
                      >
                        Password
                      </p>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                      >
                        <div
                          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1lpa638-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                          </div>
                          <input
                            aria-label="Enter your password"
                            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
                            id="password"
                            name="password"
                            placeholder="Enter your password"
                            type="password"
                            value="1234"
                          />
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                            style="cursor: pointer;"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                              data-testid="VisibilityIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                              />
                            </svg>
                          </div>
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-hdw1oc"
                            >
                              <span
                                class="notranslate"
                              >
                                ​
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-sm-10 MuiGrid-grid-md-12 css-1m8i8sn-MuiGrid-root"
                  style="margin-top: 5%;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item css-4lhqxd-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root makeStyles-checkbox-22 css-m94tiq-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                        style="font-size: 12px; margin-left: -10%; font-weight: 400;"
                      >
                        Remember Me
                      </h6>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-h6 css-oblp1g-MuiTypography-root"
                      href="/signup"
                      style="text-decoration: none; font-size: 12px; margin-left: -10%; font-weight: 400;"
                    >
                      Forget Password
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-sm-10 MuiGrid-grid-md-12 css-1exrh7x-MuiGrid-root"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-btn-20 css-sghohy-MuiButtonBase-root-MuiButton-root"
                  role="button"
                  tabindex="0"
                  to=""
                  type="submit"
                >
                  Login
                   
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item css-dcx4qa-MuiGrid-root"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
